2_Services.Tests/Base/ServiceTest.cs

```diff
@@ -0,0 +1,189 @@
+        /// </summary>
+        /// <param name="entity">The entity which is used as data to test</param>
+        public virtual void AddTest_GoodFlow(TDomain entity)
+        {
+            RepositoryMock.Setup(
+                repository => repository.Add(entity));
+
+            Service.Add(entity);
+
+            Assert.DoesNotThrow(() => {
+                RepositoryMock.Verify(repository => repository.Add(entity), Times.Once);
+                });
+        }
+
+        /// <summary>
+        /// Test if the repository method is called
+        /// </summary>
+        /// <param name="entities">The entities which are used as data to test</param>
+        public virtual void AddRangeTest_GoodFlow(IEnumerable<TDomain> entities)
+        {
+            RepositoryMock.Setup(
+                repository => repository.AddRange(entities));
+
+            Service.AddRange(entities);
+
+            Assert.DoesNotThrow(() => {
+                RepositoryMock.Verify(repository => repository.AddRange(entities), Times.Once);
+            });
+
+        }
+
+        /// <summary>
+        /// Test if the repository method is called
+        /// </summary>
+        /// <param name="entity">The entity which is used as data to test</param>
+        public virtual void Update(TDomain entity)
+        {
+            RepositoryMock.Setup(
+                repository => repository.Update(entity));
+
+            Service.Update(entity);
+
+            Assert.DoesNotThrow(() => {
+                RepositoryMock.Verify(repository => repository.Update(entity), Times.Once);
+            });
+        }
+
+        /// <summary>
+        /// Test if the repository method is called
+        /// </summary>
+        /// <param name="entity">The entity which is used as data to test</param>
+        public virtual void Remove(TDomain entity)
+        {
+            RepositoryMock.Setup(
+                repository => repository.Remove(entity));
+
+            Service.Remove(entity);
+
+            Assert.DoesNotThrow(() => {
+                RepositoryMock.Verify(repository => repository.Remove(entity), Times.Once);
+            });
+        }
+
+        /// <summary>
+        /// Test if the repository method is called
+        /// </summary>
+        /// <param name="id">The id which is used to remove the entity</param>
+        /// <returns></returns>
+        public virtual async Task RemoveAsync(int id)
+        {
+            RepositoryMock.Setup(
+                repository => repository.RemoveAsync(id));
+
+            await Service.RemoveAsync(id);
+
+            Assert.DoesNotThrow(() => {
+                RepositoryMock.Verify(repository => repository.RemoveAsync(id), Times.Once);
+            });
+        }
+
+        /// <summary>
+        /// Test if the repository method is called
+        /// </summary>
+        /// <param name="entities">The entities which are used as data to test</param>
+        /// <returns></returns>
```

Comment:
Unnecessary comment tags

Reply:
The coding guidelines require a documentation comment for each method.

Reply:
Talking about the empty <returns> tags

Reply:
The empty return tags^

Reply:
The tag itself is required according to the coding guidelines, but it may not be empty.

Reply:
The method is returning an empty Task. Setting the return in the comments in the unit tests is completely unnecessary since only nUnit will call the tests. The commentary in the unit tests is there just to explain what the test does. You will never call the test by yourself.

Reply:
Every unit test is returning an empty task or void which is both an empty return type